### PR TITLE
chore: switch to pnpm

### PR DESCRIPTION
**Pull request by christianhg**
Lots of weirdness related to running an `npm` monorepo. `pnpm` is much more straightforward and, for example, requires us to specify dependencies in the `apps` that were otherwise implicit.

### CODE REVIEW

**Review by stipsan — DISMISSED** [review]
Thanks for bravely venturing out to do this work 🙌 💖  It can be painful migrating package managers, especially on monorepos 😱 
See my PR comments for suggestions that might fix the build errors :)

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -35,14 +35,14 @@ jobs:
...
-      - run: corepack enable && npm --version && npm ci --ignore-scripts && npm audit signatures
+      - run: corepack enable && pnpm --version && pnpm install --ignore-scripts && pnpm audit
         if: ${{ steps.release.outputs.releases_created }}
       # Release Please has already incremented versions and published tags, so we just
       # need to publish the new version to npm here
-      - run: npm publish --provenance --workspaces
+      - run: pnpm -r publish
         if: ${{ steps.release.outputs.releases_created }}
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
```
[the remark this answers is not in the document]
Comment (christianhg) [reply .github/workflows/release-please.yml:51]: Thank you!

File: package.json [hunk 2]
```diff
@@ -14,9 +14,5 @@
     "turbo": "^1.10.16"
   },
   "name": "astro",
-  "packageManager": "npm@10.2.2",
-  "workspaces": [
-    "packages/*",
-    "apps/*"
-  ]
+  "packageManager": "pnpm@9.1.1"
```
[the remark this answers is not in the document]
Comment (christianhg) [reply package.json]: Thanks, I've downgraded now!

File: .github/workflows/prettier.yml [hunk 3]
```diff
@@ -18,15 +18,15 @@ jobs:
...
-      - run: corepack enable && npm --version
-      - run: npm install --ignore-scripts
+      - run: corepack enable && pnpm --version
+      - run: pnpm install --dev --ignore-scripts
       - uses: actions/cache@v4
         with:
           path: node_modules/.cache/prettier/.prettier-cache
-          key: prettier-${{ hashFiles('package-lock.json') }}
-      - run: npm run format
+          key: prettier-${{ hashFiles('pnpm-lock.json') }}
```
[the remark this answers is not in the document]
Comment (christianhg) [reply .github/workflows/prettier.yml]: Fixed 🙈

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -25,13 +25,13 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          cache: npm
+          cache: pnpm
           node-version: lts/*
-      - run: corepack enable && npm --version
```
[the remark this answers is not in the document]
Comment (christianhg) [reply .github/workflows/main.yml]: Thanks!

**Pull request merged**